unit/java_bytecode/java_bytecode_parse_generics/parse_bounded_generic_inner_classes.cpp

```diff
@@ -0,0 +1,231 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing generic classes
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_symbol.h>
+#include <testing-utils/require_type.h>
+
+#include <istream>
+#include <memory>
+
+#include <util/config.h>
+#include <util/language.h>
+#include <util/message.h>
+#include <java_bytecode/java_bytecode_language.h>
+
+SCENARIO(
+  "java_bytecode_parse_bounded_generic_inner_classes",
+  "[core][java_bytecode][java_bytecode_parse_generics]")
```

Comment:
Given that inner classes seem to throw up a few bugs in several parts of test-generator, is it worth us starting to add an additional tag to unit tests involving inner classes? e.g. `[java_bytecode_inner_classes]` ?

Reply:
Good point. Shouldn't it be `[java_bytecode_parse_generics_inner_classes]` though? Or would you make it for general inner classes too? Also, not all tests that have some inner class are focusing specifically on testing aspects of inner classes, some just use it for convenience. Should all of them be tagged then?